cloudinit/config/cc_growpart.py

```diff
@@ -570,7 +570,7 @@ def handle(name: str, cfg: Config, cloud: Cloud, args: list) -> None:
     if util.is_false(mode):
         if mode != "off":
             util.deprecate(
-                deprecated="Growpart's 'mode' key with value '{mode}'",
```

Comment:
I'm think maybe we should drop this deprecation altogether (here and in the jsonschema), since it is going to be a common tripping point.

```
python3 -c 'import yaml; print(yaml.safe_load("{mode: off}"))'
{'mode': False}
```

yet [the specification only considers `true` and `false`](https://yaml.org/spec/1.2.2/#10212-boolean) to be booleans

and our schema validation isn't any help:
```
growpart.mode:  Changed in version 22.3. Specifying a boolean ``false`` value for ``mode`` is deprecated. Use ``off`` instead.
```

Reply:
I feel like holding our more strict schema deprecation message is still desired here because we don't want to interpret YAML-interpreted boolean values long-term, and we'd prefer only discrete strings for this value type.  We deprecated this back in 22.2/22.3 (2 years ago), so this deprecation shouldn't come as a surprise. The issue is that consumers are now becoming more aware of those deprecation warnings that because of non-zero exit codes from cloud-init status. Given that python3-yaml happens to interpret the value `off` as boolean False shouldn't be a concern as long as we correctly document the desired value `'off'` that we know will be loaded as the right data type.

Reply:
> yet [the specification only considers true and false](https://yaml.org/spec/1.2.2/#10212-boolean) to be booleans

oh I see why PyYAML is doing this - it only supports the 1.1 spec, never updated to 1.2

Reply:
@holmanb interesting note on pyyaml 1.2.2 spec only stating true/false ... the pyyaml 1.1 spec though establishes other values are honored [1](https://yaml.org/spec/1.1/current.html#id864510) and [2](https://yaml.org/type/bool.html)  and I think it may be retained for backward-compat.